### PR TITLE
Restaurant

### PR DESCRIPTION
**Pull request by karthi07**
Project to practice DOM manipulation by dynamically rendering a simple restaurant homepage!  using JavaScript alone to generate the entire contents of the website!

removed stylelint Github workflow - because the project has no external CSS. and workflow throws an error that it can't find any CSS file

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        <a class="nav-link" data-id="contactBtn" href="#main"> Contact </a>
+      </li>
+    </ul>
+    <form class="form-inline my-2 my-lg-0">
+      <input class="form-control mr-sm-2" type="search" placeholder="Search" aria-label="Search">
+      <button class="btn btn-outline-success my-2 my-sm-0" type="submit">Search</button>
+    </form>
+  </div>
+</nav>
+<main class="container" id="main"></main>
```
Comment (bellom) [inline src/index.js]: Your design has absolutely no problem on its own, but however, you did not respect the guidelines. There are some lessons the guidelines wants us to learn, that's why we should follow it.

image


In other words;
**You need to append each element not hard coding them.**
kindly make use of `document.createElement` and `document.appendChild`
Example: 
`div.appendChild(Navbar)`

File: package.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{
+  "name": "restaurant_page_js",
+  "version": "1.0.0",
+  "description": "Project to practe DOM manipulation by dynamically rendering a simple restaurant homepage!  using JavaScript alone to generate the entire contents of the website!",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (bellom) [inline package.json]: We should be able to run common commands like `build`, `development` should be in `package.json` inside \`scripts'. Here is a basic example ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/restaurant_page/scripts_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622983/)) in our own case here, we are missing it so I can't run `npm run build ` to generate `dist folder`, I should be able to run `npm run build` to generate `dist folder` at my end, currently using the one created by you, it shouldn't be like that.
kindly add these two scripts `build` and `development`

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @ :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 
 
**Kindly check out my comments below;**


### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Review by Forison — CHANGES_REQUESTED** [review]
## Status:: Change required❌
# 🙋‍♂️ Karthick !!!
#### Good job on implementing the issues raised on the previous review, however, there was an issue you did not address quite well🤗 .
* [ ] It appends HTML elements to the `<div>` with `id=content` when the page is loaded (Use `document.createElement` and `document.appendChild` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/restaurant_page/append_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/restaurant-page))
- To reinterate what was mention in the previous review, your HTML element must be created with javaScript and then appended to the DOM of the elements in index.js. 
For example instead creating elements with HTML like
``` <div class="container"></div>```
 rather do it this way
```
const container = document.createElement('div');
container.classList.add('container');
```
> This project seeks to test your understanding on JS dom manipulations
## That is it kindly make this change and submit the project for a re-review😎.
#### Happy coding 👨🏼‍💻.
Addo Forison

**Comment by karthi07** [comment]
@Forison  I believe the Requirement is to keep the div with id content should be in HTML and contents of the div should be appended via JS (It is already implemented )

File: src/contact.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    eiusmod
+    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation
+    ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate
+    velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
+culpa qui officia deserunt mollit anim id est laborum.</p>
+    <a class="btn btn-outline-secondary" href="#order">Call now</a>
+  </div>
+  <div class="product-device shadow-sm d-none d-md-block"></div>
+  <div class="product-device product-device-2 shadow-sm d-none d-md-block"></div>
+</div>`;
```
Comment (shshamim63) [inline src/contact.js:15]: You are assigning HTML block in a variable in then you are using it properly. But if you check the requirement it says you have to generate each element by using JS.
image
I have searched for it and found that it is reffering to the [DOM manipulation](https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Introduction).
So if you look at this [link](https://www.w3schools.com/js/js_htmldom_nodes.asp) you will get an idea on how to generate them properly. That is what the previous reviewers trying to mention. So, please create elements by using it.

File: src/home.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    eiusmod
+    tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation
+    ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate
+    velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
+      culpa qui officia deserunt mollit anim id est laborum.</p>
+    <a class="btn btn-outline-secondary" href="#order">Order Food</a>
+  </div>
+  <div class="product-device shadow-sm d-none d-md-block"></div>
+  <div class="product-device product-device-2 shadow-sm d-none d-md-block"></div>
+</div>`;
```
Comment (shshamim63) [inline src/home.js:15]: same as above...

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      </li>
+      <li class="nav-item px-3" id="contactBtn">
+        <a class="nav-link" data-id="contactBtn" href="#main"> Contact </a>
+      </li>
+    </ul>
+    <form class="form-inline my-2 my-lg-0">
+      <input class="form-control mr-sm-2" type="search" placeholder="Search" aria-label="Search">
+      <button class="btn btn-outline-success my-2 my-sm-0" type="submit">Search</button>
+    </form>
+  </div>`;
```
Comment (shshamim63) [inline src/index.js]: same as above....

File: src/order.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+              additional content. This content is a little bit longer.</p>
+            <a href="#" class="btn btn-primary">Add to Cart</a>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+  <div class="product-device shadow-sm d-none d-md-block"></div>
+  <div class="product-device product-device-2 shadow-sm d-none d-md-block"></div>
+</div>`;
```
Comment (shshamim63) [inline src/order.js:53]: Same as above....

**Review by shshamim63 — CHANGES_REQUESTED** [review]
## :clubs: Required Changes :clubs: 
___
Hi @karthi07 :wave: ,
Awesome job, your page is simple but attractive :open_mouth: . Nav links are properly functional :sunglasses: . Webpack is doing the right job. You have generated most of the tasks that are actually needed to generate the page. But a little bit of task is blocking me to approve it right now. Let's get to it.
___
### :spades: Mandatory Changes :spades: 
___
- Please replace the HTML block code with proper JS format. I know it will take you to do a lot of things. But it will help you to ensure that you know how to deal with the big projects by following pure JS conventions. And is an easy job jor you.
___
##### Please fix the issue and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Comment by Forison** [comment]
> @Forison I believe the Requirement is to keep the div with id content should be in HTML and contents of the div should be appended via JS (It is already implemented )

The point of this project is to apply the concept of JavaScript DOM manipulation instead of equating a bunch of HTML code to a JS variable. Your page is good, but you are missing the learning objective.

**Pull request closed without merging**